applications/GeoMechanicsApplication/custom_constitutive/mohr_coulomb_with_tension_cutoff.cpp

```diff
@@ -0,0 +1,272 @@
+#include "custom_utilities/stress_strain_utilities.h"
+#include "geo_mechanics_application_variables.h"
+#include "includes/serializer.h"
+
+namespace Kratos
+{
+MohrCoulombWithTensionCutOff::MohrCoulombWithTensionCutOff(std::unique_ptr<ConstitutiveLawDimension> pConstitutiveDimension)
+    : mpConstitutiveDimension(std::move(pConstitutiveDimension)),
+      mStressVector(ZeroVector(mpConstitutiveDimension->GetStrainSize())),
+      mStressVectorFinalized(ZeroVector(mpConstitutiveDimension->GetStrainSize())),
+      mStrainVectorFinalized(ZeroVector(mpConstitutiveDimension->GetStrainSize()))
+{
+}
+
+ConstitutiveLaw::Pointer MohrCoulombWithTensionCutOff::Clone() const
+{
+    auto p_result = std::make_shared<MohrCoulombWithTensionCutOff>(mpConstitutiveDimension->Clone());
+    p_result->mStressVector          = mStressVector;
+    p_result->mStressVectorFinalized = mStressVectorFinalized;
+    p_result->mStrainVectorFinalized = mStrainVectorFinalized;
+    return p_result;
+}
+
+Vector& MohrCoulombWithTensionCutOff::GetValue(const Variable<Vector>& rThisVariable, Vector& rValue)
+{
+    if (rThisVariable == CAUCHY_STRESS_VECTOR) {
+        rValue = mStressVector;
+    } else {
+        KRATOS_ERROR << "Can't get value of " << rThisVariable.Name() << ": unsupported variable\n";
+    }
+    return rValue;
+}
+
+void MohrCoulombWithTensionCutOff::SetValue(const Variable<Vector>& rVariable,
+                                            const Vector&           rValue,
+                                            const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rVariable == CAUCHY_STRESS_VECTOR) {
+        mStressVector = rValue;
+    } else {
+        KRATOS_ERROR << "Can't set value of " << rVariable.Name() << ": unsupported variable\n";
+    }
+}
+
+SizeType MohrCoulombWithTensionCutOff::WorkingSpaceDimension()
+{
+    return mpConstitutiveDimension->GetDimension();
+}
+
+int MohrCoulombWithTensionCutOff::Check(const Properties&   rMaterialProperties,
+                                        const GeometryType& rElementGeometry,
+                                        const ProcessInfo&  rCurrentProcessInfo) const
+{
+    ConstitutiveLaw::Check(rMaterialProperties, rElementGeometry, rCurrentProcessInfo);
+
+    CheckProperty(rMaterialProperties, GEO_COHESION);
+    CheckProperty(rMaterialProperties, GEO_FRICTION_ANGLE);
+    CheckProperty(rMaterialProperties, GEO_DILATANCY_ANGLE);
+    CheckProperty(rMaterialProperties, GEO_TENSILE_STRENGTH);
+    CheckProperty(rMaterialProperties, YOUNG_MODULUS);
+    CheckProperty(rMaterialProperties, POISSON_RATIO);
+    return 0;
+}
+
+void MohrCoulombWithTensionCutOff::CheckProperty(const Properties& rMaterialProperties,
+                                                 const Kratos::Variable<double>& rVariable) const
+{
+    constexpr auto min_value = 0.0;
+    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(rVariable) || rMaterialProperties[rVariable] < min_value)
```

Comment:
Since the `min_value` is used only once, we might as well write it in the condition itself:
```suggestion
    KRATOS_ERROR_IF_NOT(rMaterialProperties.Has(rVariable) || rMaterialProperties[rVariable] < 0.0)
```